### PR TITLE
CLR-15, CLR-31: Enable socketio connection for UI

### PR DESCRIPTION
**Pull request by avirmani-miso**
Add seconds countdown.
Compute seconds for each frame using fps.

### CODE REVIEW

**Review by dpalacios-miso — DISMISSED** [review]

File: main.py [hunk 1]
```diff
@@ -62,31 +96,91 @@
         decay_rate=0.5,
         min_contour_area=5000,
         background_image=background_image,
+        fps = fps,
     )
+    if args.debug:
+        image_handler = ImageLoader()
```
Comment (gchantzialexiou) [inline main.py:102]: please remove imageLoader. It doesn't offer anything since we have a UI.

File: main.py [hunk 2]
```diff
@@ -62,31 +96,91 @@
...
+                frame = cv2.resize(frame, dim, interpolation = cv2.INTER_AREA)
+                cv2.imshow("orig", frame)
             img_cropper = ImageCropper(frame)
-            frame = img_cropper.crop((150, 320, 150, 150))
-            processor.process(frame)
-            # Wait for 1 ms for user to press 'q' key
-            if cv2.waitKey(1) & 0xFF == ord("q"):
-                break
+            # Date Sep 14, camera location changed
+            frame = img_cropper.crop((180, 300, 120, 120))
```
Comment (gchantzialexiou) [inline main.py:117]: which camera are these dimensions for?

File: main.py [hunk 3]
```diff
@@ -29,13 +44,32 @@
 # Add handlers to logger
 logger.addHandler(ch)
 logger.addHandler(fh)
+logger.disabled = True
```
Comment (gchantzialexiou) [inline main.py:47]: hmm. Why disable the logs? :/

Comment (avirmani-miso) [reply main.py:47]: Because they seem to conflict with socketio updates.

File: main.py (hunk 1 above)
Comment (avirmani-miso) [reply main.py:102]: It does. It helps in debugging.

File: main.py (hunk 2 above)
Comment (avirmani-miso) [reply main.py:117]: as I mentioned in a previous PR, these dimensions are what we read from camera and have identified ROI coordinates with. But the videos from sd card are stored at higher dimension, which is accounted for by this code to make consistent in both cases.

File: main.py [hunk 4]
```diff
@@ -26,12 +40,29 @@
...
+    emit("connection", "Cleanright Backend Connected Successfully.")
+
+
+def cleanrightloop():
     # Open the video file
-    # cap = cv2.VideoCapture("3feetLeft8feetTall0degree.mp4")
-    cap = cv2.VideoCapture("rtsp://admin:123456@10.191.3.212/H265?ch=1&subtype=0")
+    #cap = cv2.VideoCapture("rtsp://admin:123456@10.191.3.212/H265?ch=1&subtype=0")
+    # cap = cv2.VideoCapture("rtsp://admin:123456@10.191.3.225/H265?ch=1&subtype=0")
+    cap = cv2.VideoCapture("./test_videos/071635-av-1.mp4")
```
[the remark this answers is not in the document]
Comment (avirmani-miso) [reply main.py]: This code was fixed before your comment. Not sure why you saw this. Please refresh before starting PR review to avoid reviewing outdated code.

File: main.py (hunk 3 above)
Comment (gchantzialexiou) [reply main.py:47]: We don't have a logs if we disable that. What confict?

File: main.py [hunk 5]
```diff
@@ -26,12 +40,29 @@
 # Add handlers to logger
 logger.addHandler(ch)
 logger.addHandler(fh)
+# logger.disabled = True
```
[the remark this answers is not in the document]
Comment (avirmani-miso) [reply main.py]: This code was fixed before your comment. Not sure why you saw this. Please refresh before starting PR review to avoid reviewing outdated code.

File: main.py (hunk 2 above)
Comment (gchantzialexiou) [reply main.py:117]: I don't understand your answer. I am asking which of the many cameras we have in the this crop coordinates corresponds. Thanks!

File: main.py (hunk 3 above)
Comment (avirmani-miso) [reply main.py:47]: When I enabled the logger, it was not sending socketio updates. I just checked again, it seems that it may have been due to the port trouble I was running during setup. It is working with the logs now. I will remove this line of code.

File: image_loader.py [hunk 6]
```diff
@@ -25,11 +25,11 @@ def __init__(self):
     def load_images(self):
```
[the remark this answers is not in the document]
Comment (gchantzialexiou) [reply image_loader.py]: what is the integration problem with the UI? You just need to start the container

File: test_videos/071635-av-1.mp4
Comment (brugsy) [inline test_videos/071635-av-1.mp4:1]: Probably should not upload videos to github. Maybe google cloud storage for this or something.

Comment (avirmani-miso) [reply test_videos/071635-av-1.mp4:1]: yes, good catch.  I collected them separate, this one got attached. I'll remove it.

File: .gitignore [hunk 7]
```diff
@@ -163,3 +163,6 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+# Local test videos
+./test_videos
```
Comment (gchantzialexiou) [inline .gitignore:168]: what is the point of this gitignore?

Comment (avirmani-miso) [reply .gitignore:168]: So I can test locally with couple videos but they will not get uploaded to github.

Comment (avirmani-miso) [reply .gitignore]: I have removed it .

File: image_processing/handwashing_states.py [hunk 8]
```diff
@@ -50,16 +50,10 @@ def handle(self, context):
...
-            cv2.imshow(
-                "Handwashing Result", image_handler.images["handwashing_compliant"]
-            )
-        elif 150 < context._frame_count < 600:
+        elif 3 < nsecs < 20:
             logging.info("Handwashing not compliant")
-            cv2.imshow(
-                "Handwashing Result", image_handler.images["handwashing_non_compliant"]
-            )
         logging.info(f"frame count is: {context._frame_count}")
```
[the remark this answers is not in the document]
Comment (avirmani-miso) [reply image_processing/handwashing_states.py]: done.

**Review by gchantzialexiou — APPROVED** [review]
